File: go/merge_sort/merge_sort_array_test.go
package merge_sort

import (
	"github.com/stretchr/testify/assert"
	"reflect"
	"testing"
)

func TestStringRandomCases(t *testing.T) {
	for range 1_000 {
		td := newRandomStringTestData()

		actual := append([]string{}, td.array...)
		MergeSortArray(actual, func(a, b string) int {
			return len(a) - len(b)
		})

		if !reflect.DeepEqual(td.expectedArray, actual) {
			assert.FailNowf(t, "",
				"MergeSortArray(%v):\n  Actual: %v\nExpected: %v",
				td.array, actual, td.expectedArray,
			)
		}
	}
}

File: python/merge_sort/merge_sort_array_test.py
from merge_sort_array import merge_sort_array
from merge_sort_strings_test_data import RandomStringsTestData


def test_string_random_cases():
	for _ in range(1_000):
		td = RandomStringsTestData()

		got = td.array.copy()
		merge_sort_array(got, lambda a, b: len(a) - len(b))

		assert got == td.want_array, \
			f'merge_sort_array({td.array}):\nwant: {td.want_array}\n got: {got}'
